### ISSUE TITLE
Constraint involving postulate with associated rewrite rule not postponed?

### ISSUE THREAD

**Issue by nad** [agda/agda#1987]
Consider the following code:

```
{-# OPTIONS --rewriting #-}

data Unit : Set where
  unit : Unit

_+_ : Unit → Unit → Unit
unit + x = x

data _≡_ (x : Unit) : Unit → Set where
  refl : x ≡ x

{-# BUILTIN REWRITE _≡_ #-}

postulate
  f  : Unit → Unit
  fu : f unit ≡ unit

{-# REWRITE fu #-}

g : Unit → Unit
g unit = unit

data D (h : Unit → Unit) (x : Unit) : Set where
  wrap : Unit → D h x

run : ∀ {h x} → D h x → Unit
run (wrap x) = x

postulate
  d₁ : ∀ n x y (p : D y n) → x (run p + n) ≡ y (run p + n) → D x n

d₂ : ∀ s n → D s n
d₂ _ _ = wrap unit

d₃ : D (λ _ → unit) unit
d₃ = d₁ _ (λ _ → unit) (λ n → f (g n)) (d₂ (λ n → f (g n)) _) refl
```

The code is rejected:

```
unit != (f (g (run (d₂ (λ n → f (g n)) _44) + _44))) of type Unit
when checking that the expression refl has type
unit ≡ f (g (run (d₂ (λ n → f (g n)) _44) + _44))
```

However, it is accepted if `refl` is replaced with a hole, the module is loaded, and `refl` is given in the hole. The code is also accepted if `f` is replaced by `g` in the body of `d₃`.

I guess that the problem is that Agda treats `f` as an inert postulate, even though there is a possibility that the meta-variable `_44` may later be instantiated, thus triggering computation (including rewriting for `f`).

**Comment by jespercockx** [comment]
Seems to be an instance of issue #1663. I'm not sure yet why the fix in 8aaf1056f249498f09d04eeb8363cbd3a0951416 doesn't work in this case...

**Comment by jespercockx** [comment]
Fixed by f21aa28c1a5b8cad06b0f220fb6a630e4130783e
